components/zcash_note_encryption/src/lib.rs

```diff
@@ -276,6 +295,10 @@ pub trait Domain {
     /// Returns `None` if `out_plaintext` does not contain a valid byte encoding of an
     /// `EphemeralSecretKey`.
     fn extract_esk(out_plaintext: &OutPlaintextBytes) -> Option<Self::EphemeralSecretKey>;
+
+    fn separate_tag_from_ciphertext(enc_ciphertext: Self::EncNoteCiphertextBytes) -> (Self::NotePlaintextBytes, [u8; AEAD_TAG_SIZE]);
+
+    fn convert_to_compact_plaintext_type(enc_ciphertext: Self::CompactEncNoteCiphertextBytes) -> Self::CompactNotePlaintextBytes;
```

Comment:
This is cosmetic mostly, both these types would be the same length arrays in implementations. Might be worth putting this as `CompactBytes` so that switching between them happens automatically -- this is what was happening earlier, since both were just `[u8, COMPACT_NOTE_SIZE]` then.